### PR TITLE
Adding a working code example for plotting intermediate values.

### PR DESCRIPTION
**Pull request by barneyhill**
I replaced the existing code example with an example of gradient descent. This allows for intermediate values to be recorded and plotted. The issue is mentioned at [#1033](https://github.com/optuna/optuna/issues/1033)

### CODE REVIEW

File: optuna/visualization/intermediate_values.py [hunk 1]
```diff
@@ -25,13 +25,27 @@ def plot_intermediate_values(study):
 
             import optuna
 
+            # Derivative function for x**2
```
Comment (hvy) [inline optuna/visualization/intermediate_values.py]: Ditto.
```suggestion
            # Derivative function for x ** 2
```

File: optuna/visualization/intermediate_values.py [hunk 2]
```diff
@@ -25,13 +25,27 @@ def plot_intermediate_values(study):
...
+
+                next_x = 1  # We start the search at x=1
+                gamma = trial.suggest_loguniform('alpha', 1e-5, 1e-1)  # Step size multiplier
+
+                # Stepping through gradient descent to find the minima of x**2
+                for step in range(100):
+                    current_x = next_x
+                    next_x = current_x - gamma * df(current_x)
+
+                    delta = next_x - current_x
```
Comment (hvy) [inline optuna/visualization/intermediate_values.py]: How about defining the function "f", i.e. `x ** 2` and report/return the value of "f" given the current x?

**Comment by hvy** [comment]
@barneyhill Sorry I don't mean to stress but do you have any updates on the status of this PR, or would you like me to take over?

**Comment by barneyhill** [comment]
Hey @hvy sorry I've been very busy and should of said something. I spent a couple of hours today trying to get circleCI working but I'm still unable to get it to run. Is it okay if you take this PR over?

**Comment by hvy** [comment]
@barneyhill I created a PR to address the same issue https://github.com/optuna/optuna/pull/1103. Maybe I could help you out with the CI troubles if they reoccur in your next PR when you have more time.

**Comment by hvy** [comment]
Let me close this PR as it's taken over by #1103. Thanks @barneyhill for the effort and addressing the issue.

**Pull request closed without merging**